crates/pathfinder/src/rpc/api.rs

```diff
@@ -276,7 +279,7 @@ impl RpcApi {
             .await?;
         // The default value should actually never happen as block_number is None only
         // in a pending block.
-        Ok(block.block_number.unwrap_or_default())
+        Ok(block.block_number.unwrap_or_default().0)
```

Comment:
If this shouldn't happen then isn't this an error?

Reply:
This is a consequence of having common structs for normal and pending blocks and at the same time the rpc spec not specifying these fields as optional. At the moment I assumed zero hash for such values (ie. missing values), whereas if a value can fail during conversion, I will just bail with a proper error (already discussed with Joonas).
Maybe I should ask Starkware how the pending blocks should be represented in the rpc api :thinking: .

Reply:
You should bail out with an error since this is data coming from outside our control. Once (if) the pending stuff is separated out then this would still return a parsing error since a required field is missing. So we do the same now.

Reply:
Err() case added.